src/main/java/org/fenixedu/academic/domain/organizationalStructure/UnitName.java

```diff
@@ -223,44 +210,13 @@ private static void addDifferentUnitNames(final int size, final UnitNameLimitedO
 
     private static boolean containsAllExactWords(final String normalizedUnitName, final String[] nameParts) {
         final String[] unitNameParts = UnitNamePart.getNameParts(normalizedUnitName);
-        for (final String namePart : nameParts) {
-            if (namePart.length() > 3) {
-                if (!existsCompleteNamePart(unitNameParts, namePart)) {
-                    return false;
-                }
-            }
-        }
-        return true;
+        return Arrays.stream(nameParts)
+                .filter(namePart -> namePart.length() > 3)
+                .allMatch(namePart -> existsCompleteNamePart(unitNameParts, namePart));
     }
 
     private static boolean existsCompleteNamePart(final String[] unitNameParts, final String namePart) {
-        for (String unitPart : unitNameParts) {
-            if (unitPart.equalsIgnoreCase(namePart)) {
-                return true;
-            }
-        }
-        return false;
-    }
-
-    public static void findExactWords(final UnitNameLimitedOrderedSet unitNameLimitedOrderedSet, final String name) {
-        final String[] nameParts = UnitNamePart.getNameParts(name);
-        if (nameParts.length > 0) {
-            final UnitNamePart unitNamePart = UnitNamePart.find(nameParts[0]);
-            if (unitNamePart != null && nameParts.length == 1) {
-                unitNameLimitedOrderedSet.addAll(unitNamePart.getUnitNameSet());
-            } else {
-                final Set<UnitName> unitNames =
-                        unitNamePart == null ? Bennu.getInstance().getUnitNameSet() : unitNamePart.getUnitNameSet();
-                for (final UnitName unitName : unitNames) {
-                    final String normalizedUnitName = unitName.getName();
-                    if (containsAllExactWords(normalizedUnitName, nameParts)) {
-                        if (unitName.getUnit().getCode() != null) {
```

Comment:
This line here garantees that the units sent in the RAIDES context have a code, as they must, removing this will make the information not compatible with what is expected by DGES.
This filter must be applied when searching for school units that are going to be used for RAIDES reports

Reply:
Ok but that requirement is already satisfied in the overriden add function of ExternalAcademicUnitNameLimitedOrderedSet: 

https://github.com/FenixEdu/fenixedu-academic/blob/7fc51aac48ac91e6af2d156374b96fdb58348cfa/src/main/java/org/fenixedu/academic/domain/organizationalStructure/UnitName.java#L128-L132

Since the && operator uses short-circuit evaluation and there is StringUtils.isEmpty validation, invalid units are never added to the set (done by the super call).